packages/grpc-native-core/ext/channel.cc

```diff
@@ -88,10 +110,31 @@ bool ParseChannelArgs(Local<Value> args_val,
     } else if (value->IsString()) {
       Utf8String val_str(value);
       channel_args->args[i].type = GRPC_ARG_STRING;
-      channel_args->args[i].value.string =
-          reinterpret_cast<char *>(calloc(val_str.length() + 1, sizeof(char)));
-      memcpy(channel_args->args[i].value.string, *val_str,
-             val_str.length() + 1);
+      /* Append the grpc-node user agent string after the application user agent
```

Comment:
Maybe we want to add some tests for this codepath ? Like, what we should expect as a user agent string when we give no user agent, and when we give one.

Reply:
We do have [this test](https://github.com/grpc/grpc-node/blob/master/packages/grpc-native-core/test/surface_test.js#L487). I could add another one to check the case of passing the option.